apps/examples/connect_test/connect_test_server.c

```diff
@@ -0,0 +1,815 @@
+
+	snprintf(buf, HOSTNAME_BUFSIZE, "%s\n", ap_info->hostname);
+	file_name = ap_info->path;
+
+	ret = _wm_store_file(buf, HOSTNAME_BUFSIZE, file_name);
+	if (ret < 0) {
+		CT_LOGE(TAG, "store file failed");
+		return -2;
+	}
+	return 0;
+}
+
+void connect_test_add_hostname(void *arg)
+{
+	struct ct_options *hostname_info = (struct ct_options *)arg;
+	int ret = -1;
+	CT_LOG(TAG, "Hostname[%s]", hostname_info->hostname);
+	ret = _wm_save_hostname(hostname_info);
+	if (ret < 0) {
+		CT_LOGE(TAG, "Save AP configuration failed");
+		return;
+	}
+}
+
+static void hostname_to_ip4(char *hostname)
+{
+        if (hostname == NULL) {
+                CT_LOGE(TAG, "Hostname not present");
+		return;
+        }
+
+        struct hostent *shost;
+        shost = (struct hostent*)gethostbyname((const char *)hostname);
+        if (shost == NULL) {
+                CT_LOGE(TAG, "gethostbyname fail %d", shost);
+                return;
+        }
+        CT_LOG(TAG, "Hostname: %s IPAddress: %s", hostname,
+		inet_ntoa(*((struct in_addr *)shost->h_addr_list[0])));
+}
+
+static void default_dns_service(void) {
+	CT_LOG(TAG, "Running default DNS service");
+	char hostname[30]; 
+	char hostnames[3][20] = {"www.google.com", "www.samsung.com", "www.facebook.com"};
+	for (int hname = 0; hname < 3; hname++) {
+		strcpy(hostname, hostnames[hname]);
+		hostname_to_ip4(hostname);
+	}
+}
+
+static int dns_service(void) {
+	if (!g_WM_HOSTNAME_FILE) {
+		default_dns_service();
+	} else {
+		char *file_name = g_WM_HOSTNAME_FILE;
+		FILE* file = fopen(file_name, "r"); 
+		if(!file) {
+			CT_LOG(TAG, "Unable to open : %s", file_name);
+			default_dns_service();
+		} else {
+			char hostname[30]; 
+			while (fgets(hostname, sizeof(hostname), file)) {
+				hostname[strlen(hostname) - 1] = '\0';
+				hostname_to_ip4(hostname); 
+			}
+			fclose(file);
+		}
+	}
+	return 0;
+}
+
+//UDP Data Transfer
+static int _recv_data_udp(int fd, int size, struct sockaddr_in *caddr) 
+{
+	char buf[WT_BUF_SIZE];
+	int remain = size;
+	unsigned int clen = sizeof(struct sockaddr_in);
+	int cnt = 1;
+	int total_received = 0;
+	CT_LOG(TAG, "_recv_data_udp start");
+	while (remain > 0) {
+		int read_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = recvfrom(fd, buf, read_size, 0, (struct sockaddr*)caddr, &clen);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_received += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+	return 0;
+}
+
+static int _send_data_udp(int fd, int size, struct sockaddr_in *caddr)
```

Comment:
server and client use similar features. how about to remove repeated code by separating it to the file.

Reply:
changes done as suggested